Add Docker image

RyanGlScott opened the pull request:
Fixes #71.

RyanGlScott commented:
I have verified that this image builds locally and runs as you would expect it to. Some things left to do:

* This image does _not_ build our forked versions of the Rust standard libraries (currently housed [here](https://github.com/GaloisInc/crucible/tree/90447535c05834fcb2bd7bb0aa64847b788f69bf/crux-mir/lib)). After some thought, I think this is fine. The plan in https://github.com/GaloisInc/crucible/issues/1252 is to move this code to its own repo, and once we do that, we can provide a separate Docker image that includes the compiled versions of the forked standard libraries.
* We should build this image as part of CI and push it to the GitHub Container Registry (ghcr.io). For now, I think it will suffice to do this on each commit to the `master` branch (where the image is tagged with `nightly`), as well as on each time a `git` tag is pushed (where the image is tagged with the name of the `git` tag).
* We should advertise this image in the `mir-json` `README`.

spernsteiner commented:
> This image does _not_ build our forked versions of the Rust standard libraries

Is mir-json usable for anything without those libraries?  Even `#[no_std]` code is going to need libcore, and using the standard libcore (e.g. `cargo saw-build -Zbuild-std=core`) would be pretty severely broken (IIRC even some basic slice operations are patched in our fork).

RyanGlScott commented:
> Is mir-json usable for anything without those libraries?

No, not really. (And especially so after https://github.com/GaloisInc/mir-json/issues/67 is fixed.)

I suppose there is a question of whether we want the `mir-json` repo to house a dev image that provides nothing but `mir-json`, for which the repo that houses the fork of the Rust standard libraries could use to build its own image. Alternatively, we could not offer an image in the `mir-json` repo at all and instead provide an image in the other repo that includes all of `mir-json` + the forked standard libraries. I don't have a particularly strong opinion on how we do it.

Yet another alternative would be to move the forked standard libraries into this repo so that both them and `mir-json` are all under the same repo.

spernsteiner commented:
I have no real preference, except that I don't think we should advertise an image to users that they can't actually use for anything

RyanGlScott commented:
I have rebased this on top of the changes from https://github.com/GaloisInc/crucible/issues/1252, and I have updated the Docker image to include the translated standard libraries. Next up: a CI job.

qsctr commented on .dockerignore:
```diff
@@ -0,0 +1,25 @@
+.git
+.gitignore
+
+# Artifacts of running translate_libs.hs
```
```suggestion
# Artifacts of running translate_libs.sh
```

qsctr reviewed (approved):
I am not a CI expert but this looks ok to me
